commit 1782643131d5bbbfc54827402970e70e4e755c71
Author: Kalesh Singh <kaleshsingh@google.com>
Date:   Tue Oct 15 12:23:23 2024 -0700

    ltp: syscalls: pipe2: Update pipe size check
    
    If the pipe size is set to a value less than the page size. The
    kernel rounds the size up to 1 page.
    
    In the case of Android's x86_64 16KiB emulator this is rounded up
    to the 4KiB kernel page size.
    
    Update the test to handle this case. This is a no-op on all other
    targets.
    
    Bug: 366098040
    Test: atest vts_ltp_test_x86_64:syscalls.pipe2_04_64bit#syscalls.pipe2_04_64bit
    Change-Id: I78fce55fef53c0f628fc5aa5789fddd7f16fbefb
    Signed-off-by: Kalesh Singh <kaleshsingh@google.com>

diff --git a/testcases/kernel/syscalls/pipe2/pipe2_04.c b/testcases/kernel/syscalls/pipe2/pipe2_04.c
index 3911f95b4..40ab005cf 100644
--- a/testcases/kernel/syscalls/pipe2/pipe2_04.c
+++ b/testcases/kernel/syscalls/pipe2/pipe2_04.c
@@ -15,6 +15,7 @@
 #include <stdio.h>
 #include "lapi/fcntl.h"
 #include "tst_test.h"
+#include "pgsize_helpers.h"
 
 static int fds[2];
 static int flags;
@@ -61,7 +62,7 @@ static void setup(void)
 	char *write_buffer;
 
 	SAFE_PIPE2(fds, O_NONBLOCK);
-	page_size = SAFE_SYSCONF(_SC_PAGESIZE);
+	page_size = kernel_page_size();
 
 	flags = SAFE_FCNTL(fds[1], F_GETFL);
 	if (!(flags & O_NONBLOCK))
